plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -865,12 +865,23 @@ private static void cleanExtraOutputFiles(TrinoFileSystem fileSystem, String que
             log.info("Found %s files to delete and %s to retain in location %s for query %s", filesToDelete.size(), fileNamesToKeep.size(), location, queryId);
             ImmutableList.Builder<String> deletedFilesBuilder = ImmutableList.builder();
             Iterator<String> filesToDeleteIterator = filesToDelete.iterator();
+            List<String> deleteBatch = new ArrayList<>();
             while (filesToDeleteIterator.hasNext()) {
                 String fileName = filesToDeleteIterator.next();
-                log.debug("Deleting failed attempt file %s/%s for query %s", location, fileName, queryId);
-                fileSystem.deleteFile(location + "/" + fileName);
+                deleteBatch.add(location + "/" + fileName);
                 deletedFilesBuilder.add(fileName);
                 filesToDeleteIterator.remove();
+
+                if (deleteBatch.size() >= DELETE_BATCH_SIZE) {
```

Comment:
keep `deleteBatch.add(location + "/" + fileName);` and `if (deleteBatch.size() >= DELETE_BATCH_` lines together